ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/BlockingQueryPublisher.java

```diff
@@ -90,9 +102,23 @@ public void setQueryHandle(final QueryHandle queryHandle, final boolean isPullQu
     // we should be returning a "Limit Reached" message as we do in the HTTP/1 endpoint when
     // we hit the limit, but for query completion, we should just end the response stream.
     this.queue.setCompletionHandler(() -> {
+      if (isPullQuery) {
```

Comment:
First, this code change is quite unrelated to this PR :)  Second, I think we should keep it since CP is adding `LIMIT` support for pull queries and we would want the completion to be handled differently, right?

Reply:
The limit stuff is already merged :) but feel free to go about it either way